### DISCUSSION TITLE
Shield Rebalancing

### DISCUSSION THREAD

**Discussion by PFSCommissar** [tsunamayo/Starship-EVO#4642] [Suggestions]
I recently decided to sit down and do an in-depth analysis of shield strategies, asking the question: should I invest more in shield recharge time, or shield capacity? 

Shield Recharge Time will hereafter be referred to as SRT, and Shield Hit Points will be SHP

To do this analysis, I wrote a MATLAB code that evaluates the time-to-kill based on the weapon DPS, the SRT, and SHP. This code uses the assumptions that DPS will be constant, and SRT is the listed recharge time, plus 4 seconds. This is consistent with my in-game observations that recharge will begin 4s after shooting stops, and then it will take an amount of seconds equal to the SRT to regenerate, regardless of if shooting continues or not. 

The code then returns the amount in seconds that it will take to kill the ship for two cases: 1) where 100% of shield block space are shield blocks (i.e. minimum SRT possible), and 2) where a certain percentage of that block space was used instead for shield capacitors. The final result is the difference in the two time-to-kills, showing which one is superior.

What I had hoped to find was a curve where there are optimal configurations and it would be easy to prefer one or the other depending on the case. What I found instead was a bit concerning:

Short Version: **Capacity should always be maxed** except in cases where the DPS hitting you is very low, around 0.0375*your SHP

Here's a low DPS profile:
In the graphs below, the Y axis is the time to kill margin in seconds. A positive number means that emphasizing capacity is preferable over emphasizing recharge time, and negative is vice versa. The x axis is the fraction of the blocks given over to capacitors, so at 0.9, 90% of the shield blocks have been traded out for capacitors. 

![image](https://user-images.githubusercontent.com/67161181/159696883-570a5089-6b73-48ce-9d8a-649f385f1179.png)
You can see here that when the DPS is very low, there are cases where recharge time is superior, but here we're talking about 0.0375*SHP, which is 1500 DPS vs a 40000 HP shield. When the Hull HP are also 40k, this means that the minimum time to kill is 133 seconds. The possibility of any low DPS ship surviving that long is very low, so I would argue that this low DPS case should not even be considered in the data.

as we move upwards in DPS, maximizing SHP (capacity) is always preferable:
![image](https://user-images.githubusercontent.com/67161181/159697424-8128d8e7-3108-4a31-ba1e-564ad8faf6d2.png)
This is for 3000 DPS, which is 0.075 of the total SHP (40000)

once we hit 0.1*SHP (4000 DPS vs 40000 SHP), the trend is obvious and calls for capacity to always be maximized:
![image](https://user-images.githubusercontent.com/67161181/159697947-47f45926-f732-418b-86f3-e303493be910.png)

Here's another way to view the data:
In these graphs below, the Shield Capacitor % is held constant, while the DPS is calculated as % of hull HP and increased across the x-axis
below is the 90% case, where 90% of shield blocks are traded for capacitors. While it's obvious there are diminishing returns as DPS increases, note how the Y axis is always positive. Therefore, **it is always preferable to maximize capacitors**
![image](https://user-images.githubusercontent.com/67161181/159698634-c63fe188-ce2d-43b0-ad59-bb94b22938e6.png)

you'll get some negative margins (meaning SRT is preferable) if you use a low cap %. This is a 10% cap case:
![image](https://user-images.githubusercontent.com/67161181/159699120-f7801db6-ef6d-4223-8121-663e459b3371.png)
However, notice that the margins are much smaller in magnitude, meaning that the 90% cap case, which is superior for all DPS percentages, is superior to this 10% case. 

because DPS and Shield Capacitors are calculated by %, the value of your Hull HP and Shield HP do not matter, so the trends will be consistent for every ship and the margins will remain the same. 

As a final point, shield capacitors do not come with the power cost that shield blocks do, giving the final nail in the coffin to shield blocks. So even if you think you'll want shield blocks and SRT for those super low DPS cases, the cost to your ship's power may not be worth it. **There is therefore no compelling reason to not maximize capacity.** 

In my opinion, this is a balancing problem. All other ship parameters come at a cost, and are balancing game of give and takes, as real life engineering is. However, what is described above is a case where one should always do the same strategy of maximizing in every case with no need to optimize. This takes away from the engineering element involved, as there is now no need for thought or analysis when deciding how you want to do your shields. 

I would suggest three options as a solution to this: 
1) make capacitors have bigger drawbacks. this could be in heat cost, power cost, or increase recharge time more than they already do. there should come a point where it is no longer preferable to choose them, but right now that's not the case
2) the preferable option: rework shield recharge mechanics: i would suggest an "active recharge" where shields are constantly recharging, regardless of whether they are being shot at or not. this option probably requires more work and rebalancing, but i'm happy to lend analysis to it as i think it could lend to some very interesting options where we have active shield tanks vs passive ones. capacitors would lower SRT just like they do now, and this would likely present some cases where it is much better to have a shield that instantly heals the very low DPS it receives (and is therefore invulnerable to it) than one that is very strong but can be whittled away at by tiny weapons. With this option, therefore, that Low DPS range where it's preferable to have SRT will widen and give it more tactical viability. 
3) this option could come alongside either of the first two options: add blocks like capacitors, but that increase recharge rate instead for some very fast recharging shields.

**Comment by Briaireous** [comment]
O gawd, never thought I'd hear about MATLAB here

**Reply by PFSCommissar** [reply]
i spent too much time doing spaceflight dynamics projects with MATLAB to just let all that completely job-relevant skill go to waste after graduating
